### ISSUE TITLE
Support proper initialization order for static SDS clusters 

### ISSUE THREAD

**Issue by qiwzhang** [envoyproxy/envoy#4522]
*Title*: Support proper initialization order for static SDS clusters 

*Description*:
If a static cluster is using SDS (fetching certificates from remote), and it is using envoy_grpc which requires another static sds cluster.  It needs to initialize sds cluster first before initializing the cluster using sds.

**Comment by junr03** [comment]
@qiwzhang what is the current behavior. Will the cluster using SDS retry getting certificates from the remote, so that eventually after the SDS cluster is setup the cluster using SDS will succeed?

**Comment by qiwzhang** [comment]
Yes,  subscription will retry right away after the first failure.  the cluster of using SDS will get the certificate. I even have an integration test to test this case.  https://github.com/envoyproxy/envoy/blob/master/test/integration/sds_dynamic_integration_test.cc#L367
